src/main/java/org/jabref/gui/maintable/OpenExternalFileAction.java

```diff
@@ -24,32 +25,53 @@ public OpenExternalFileAction(DialogService dialogService, StateManager stateMan
         this.stateManager = stateManager;
         this.preferencesService = preferencesService;
 
-        this.executable.bind(ActionHelper.isFilePresentForSelectedEntry(stateManager, preferencesService)
-                                         .and(ActionHelper.needsEntriesSelected(1, stateManager)));
+        this.executable.bind(ActionHelper.hasPresentFileForSelectedEntries(stateManager, preferencesService)
+                .and(ActionHelper.needsEntriesSelected(stateManager))
+        );
     }
 
+    /**
+     * Open all linked files of the selected entries.
+     * <br>
+     * If some selected entries have linked file and others does not, pop out a dialog to ask user whether to skip them and continue or cancel the whole action.
+     */
     @Override
     public void execute() {
         stateManager.getActiveDatabase().ifPresent(databaseContext -> {
             final List<BibEntry> selectedEntries = stateManager.getSelectedEntries();
 
-            if (selectedEntries.size() != 1) {
-                dialogService.notify(Localization.lang("This operation requires exactly one item to be selected."));
-                return;
+            List<LinkedFileViewModel> linkedFileViewModelList = new LinkedList<>();
+            LinkedFileViewModel linkedFileViewModel;
+
+            boolean asked = false;
+
+            for (BibEntry entry:selectedEntries) {
+                if (entry.getFiles().isEmpty()) {
+                    if (!asked) {
+                        boolean continu = dialogService.showConfirmationDialogAndWait(Localization.lang("Missing file"),
```

Comment:
I, personally, don't think it's necessary to present a dialog here to the user. To me it seems obvious that only the files will open from entries that have associated files 
So just ignore them.